docs/program/apis/sessions.md

```diff
@@ -0,0 +1,143 @@
+---
+title: "Session Management with Viam's Client SDKs"
+linkTitle: "Session Management API"
+weight: 20
+type: "docs"
+description: "How to use the session management API with Viam's Client SDKs."
+tags: ["client", "sdk", "viam-server", "networking", "apis", "robot api", "session", "sessions", "session management"]
+---
+
+The session management API provides support for robot session management.
+
+A session:
+
+- allows a client to express that it is actively connected or authenticated to `viam-server` on your robot
+- supports stopping moving components when the connection is no longer active.
+
+### Purpose of session management
+
+When controlling a robot or fleet with Viam, you want a way to understand the presence of the clients that are communicating with and authenticated to each robot's `viam-server`.
+
+The period of time during which these clients are present is called a *session*.
+*Session management* is a presence mechanism between a client and `viam-server`.
+
+Session management allows for safer operation of robots that physically move.
+For example, imagine a wheeled rover gets a [`SetPower()`](/components/base/#setpower) command as the last input from a client before the connection to the robot is interrupted.
+Without session management, the API request from the client sets the flow of electricity to the motors of the robot and then does not time out, causing the robot to continue driving forever, colliding with objects and people.
+
+With session management, if the robot does not receive a signal from the client at regular intervals, it safely stops until the connection is reestablished.
+
+#### Clients and Viam's client SDKs
+
+*Client* has multiple meanings for a Viam robot.
+Essentially, it's anything that is receiving the information served by `viam-server` running on the robot.
+
+A **client** can be an SDK script controlling the robot, an input controller, or just the different resources on the robot talking amongst themselves.
+For example, if you use Viam's module registry to [add modular resources to your robot](/extend/modular-resources/), the clients of your robot in its lifetime will include the "model servers" you instantiate on your robot for individual resources.
+
+Viam's session management API is your built-in solution to manage this.
```

Comment:
```suggestion
```
I realize you've added the word "manage" but "this" is still ambiguous and I still don't think this is a useful sentence. What is it meant to convey?

Reply:
"this" is the problem i'm talking about beforehand!

Reply:
The problem being that client can mean lots of different things? That's what this is currently implying, but I don't think that's what it's meant to mean.

Reply:
client can mean lots of different things, yes, but there is only one session in question for the session management api. The session is what's being discussed, not the client

Reply:
I'll try and address your feedback but I'm sorry bc I too am getting confused 😆

Reply:
@JessamyT  would you suggest the specific noun be maybe something like "this dissonance in your session"?

Reply:
Then what is the actual problem this solves? This section (under the client heading) just talks about how "client" can mean various things. Then it says "here's the solution to this problem," but what is the problem?

Reply:
I don't think I understand what "dissonance in your session" actually means, but yeah something more specific than "this" is what I'm advocating for! Could talk synchronously about this maybe :D I'll slack you

Reply:
```suggestion
Viam's session management API provides this functionality.
```
Okay, so, this will make sense if you move the "what is a client" type stuff (cut down to 2 sentences I think) into an info box before the purpose stuff. Then the "why" will flow directly into the "how".

Reply:
I think it makes sense as-is, is that ok? the this in question is the abundance of clients i'm discussing, then i go into the how of how it works.

Reply:
as-is after implementing your other suggestions!! @JessamyT

Reply:
But again, I thought the problem is *not* the abundance of clients, but rather the potential for disconnection without timeout?

Reply:
The problem that's being solved here is wanting to turn off Automated Session Management (eric's session management client), ao that all calls to/through that API you'll now make yourself instead of viam/cloud run doing that for you. it's really an advanced use case